jwst/photom/ifu_mrs.py

```diff
@@ -0,0 +1,145 @@
+import logging
+import numpy as np
+from scipy.interpolate import RegularGridInterpolator
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def expfunc_bounded(x, a, b, c, x0):
+    """ Function using time dependent coefficients """
+
+    temp = a * np.exp(-b * (x - x0) / 100) + c
+    temp[temp > 1] = 1
+    return temp
+
+
+def get_correction_function(side, timecoeff, mid_time):
+    """ Find the time and wavelength dependent function """
+
+    binwave = timecoeff[side]['binwave']
+    a = timecoeff[side]['acoeff']
+    b = timecoeff[side]['bcoeff']
+    c = timecoeff[side]['ccoeff']
+    x0 = timecoeff[side]['x0']
+
+    # time depenpent function
+    corgrid = expfunc_bounded(mid_time, a, b, c, x0)
+    # now fold in the wavelength dependence
+    func = RegularGridInterpolator([binwave], corgrid,
+                                   bounds_error=False, fill_value=None)
+    return func
+
+
+def time_correction(input, detector, ftab, mid_time):
+    """
+    Find the time and wavelength dependent photom correction for MRS data.
+
+    Parameters
+    ----------
+    input : JWST data model
+        Input science data model to be corrected.
+
+    detector : string
+        MRS detector working on
+
+    ftab : stdatamodels.jwst.datamodels.MirMrsPhotomModel
+        MRS PHotom reference file
+
+    mid_time: float
+        Exposure mid time in MJD
+
+    Returns
+    -------
+    result : numpy array
+        An array of corrections to apply to data.
+    """
+
+    # Tables to be read in from MRS Photom reference file
+    # Read in the time coefficients for each channel
+    table_ch = {}
+    table_ch['ch1'] = ftab.timecoeff_ch1
+    table_ch['ch2'] = ftab.timecoeff_ch2
+    table_ch['ch3'] = ftab.timecoeff_ch3
+    table_ch['ch4'] = ftab.timecoeff_ch4
+
+    # Each MIRI MRS exposure has 2 channels
+    # We want to correct the channels seperately
+    # If we have MIRIFUSHORT data then channel 1 is on the left and channel 2
+    # is on the right. If we have MIRIFULONG  data then channel 4 is on the left
+    # and channel 3 is on right.
+
+    timecoeff = {}
+    timecoeff['left'] = {}
+    timecoeff['right'] = {}
+    left = 'ch1'
+    right = 'ch2'
+    timecoeff['left']['xstart'] = 0
+    timecoeff['left']['xend'] = 512
+    timecoeff['right']['xstart'] = 513
+    timecoeff['right']['xend'] = 1031
+    if detector == 'MIRIFULONG':
+        left = 'ch4'
+        right = 'ch3'
+        # Check the reference file has the time dependent coefficients
+        # check that table 1 wavelength bin is an array with values
```

Comment:
Try using the `getarray_noinit()` method on the input model to see if the coeff arrays exist. What happens if they don't exist? Do you set all the coeffs to suitable defaults, such as zero or one, so that the correction factors are all 1?